binder/src/test/java/io/grpc/binder/SecurityPoliciesTest.java

```diff
@@ -382,14 +383,64 @@ public void testAllOf_succeedsIfAllSecurityPoliciesAllowed() throws Exception {
 
   @Test
   public void testAllOf_failsIfOneSecurityPoliciesNotAllowed() throws Exception {
+    final AtomicBoolean policyCalled = new AtomicBoolean(false);
     policy =
         SecurityPolicies.allOf(
             SecurityPolicies.internalOnly(),
-            SecurityPolicies.permissionDenied("Not allowed SecurityPolicy"));
+            SecurityPolicies.permissionDenied("Not allowed SecurityPolicy"),
+            new SecurityPolicy() {
+              @Override
+              public Status checkAuthorization(int uid) {
+                policyCalled.set(true);
+                throw new AssertionError();
```

Comment:
I'm struggling to understand this throw in light of the assertThat(policyCalled.get()).isFalse() assertion below. Would one assertion or the other be sufficient or is there something I'm not seeing? Same question for the second instance of this pattern below.

Reply:
The method has to return something, so throwing made more sense to me than returning a real Status. I avoided relying purely on the AssertionError in order to protect against the scenario where the code under test posts this method to another thread and/or swallows the exception.

It's purely defensive, having seen tests pass when they shouldn't under similar circumstances. I don't feel strongly about keeping it this way. Simply throwing AssertionError would be adequate given that everything is inline, currently.

I would also remove these checks altogether if we remove this guarantee from the javadoc, per the thread above.

Reply:
Thanks for explaining! I found this throw distracting from the goal of this test because the SecurityPolicy contract doesn't specify what happens if checkAuthorization() throws (especially a java.lang.Error, something "a reasonable application should not try to catch") What do you think of keeping just the final assertThat(policyCalled.get()).isFalse() check and had each unreachable SecurityPolicy return a Status consistent with the expected result of the aggregate SecurityPolicy (non-OK status here where you expect access denied anyway and Status.OK in the anyof test where you expect an access granted result) ?

Reply:
OK re-reading I see now that you want to guard against the SUT posting this method to another thread. That could still race against the whole test suite completing successfully but then at least the AssertionError() would make the test flaky ... ? Not too important what you have seems fine.